x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.test.tsx

```diff
@@ -80,13 +86,55 @@ jest.mock('react-router-dom', () => {
     ...originalModule,
     useParams: jest.fn(),
     useHistory: jest.fn(),
+    useLocation: jest.fn().mockReturnValue({ pathname: '/alerts' }),
   };
 });
 
-jest.mock('../../../../../common/lib/kibana');
-
 const mockRedirectLegacyUrl = jest.fn();
 const mockGetLegacyUrlConflict = jest.fn();
+jest.mock('../../../../../common/lib/kibana', () => {
+  const originalModule = jest.requireActual('../../../../../common/lib/kibana');
+  return {
+    ...originalModule,
+    useKibana: () => ({
+      services: {
```

Comment:
I believe you can wrap your component with `<TestProviders>` instead of managing these larger mocks manually